### DISCUSSION TITLE
添加忽略错误的语法糖

### DISCUSSION THREAD

**Discussion by mantou132** [JSCIG/es-discuss#10] [Ideas]
例如：
```js
Promise.reject()?
```
相当于：
```js
Promise.reject().catch(() => {})
```

一般来说所有的 Promise 都要处理 rejected 的情况，Node.js, Deno 遇到没有处理的 rejected Promise 甚至会退出进程。但很多时候，一些 Promise 没必要处理，这样就会出现很多多余的代码 `catch(() => {})`

另外也能替代一些 try...catch：

```js
execError()?
```
相当于 ：
```js
try {
  execError()
} catch {}
```

**Comment by haoqunjiang** [comment]
但是这个用一个 utility function 就能解决啊？比如这个被无数项目引用的 https://www.npmjs.com/package/nice-try

**Comment by Jack-Works** [comment]
如果能做到 rust 那种的 checked exception 设计 + ? 语法糖自动解包就很爽了。

**Comment by mantou132** [comment]
突然想到还是只能忽略 rejected promise。

代替 try...catch 是不可能了。try...catch 是作用在语句上的，不是一个操作符可以代替的

**Comment by chenyong** [comment]
刚好 `execError()` 返回是一个对象包含一个 `.value` 属性, 然后也可能返回 `null`, 就需要用 optional... 然后...

```js
execError()??.value
```

**Comment by Alan-Liang** [comment]
@chenyong 让我想起来了这个：
```javascript
a()??b()?+c:d
```
是指(没有这个提案时候的情况) Nullish coalescing：
```javascript
a() ?? b() ? +c : d
```
还是：
```javascript
(a()?) ? b()?+c : d
```
？

**Comment by Alan-Liang** [comment]
<https://github.com/isiahmeadows/proposal-try-expression>?
